src/core/src/runtime/tensor_utils.cpp

```diff
@@ -0,0 +1,160 @@
+        OPENVINO_ASSERT(alignment == alignof(max_align_t) || alignment == 0);
+        OPENVINO_ASSERT(!allocated);
+        OPENVINO_ASSERT(bytes == buffer->size());
+        allocated = true;
+        return buffer->get_ptr();
+    }
+
+    void deallocate(void* handle, const size_t bytes, const size_t alignment) {}
+
+    bool is_equal(const StaticBufferAllocator&) const {
+        return true;
+    }
+};
+}  // namespace
+
+void save_tensor_data(const ov::Tensor& tensor, const std::filesystem::path& file_name) {
+    OPENVINO_ASSERT(tensor.get_element_type() != ov::element::string);
+    const char* data = reinterpret_cast<const char*>(tensor.data());
+    std::ofstream fout(file_name, std::ios::binary);
+    fout.write(data, tensor.get_byte_size());
+}
+
+void read_tensor_data(const std::filesystem::path& file_name, Tensor& tensor, size_t offset) {
+    OPENVINO_ASSERT(tensor.get_element_type() != ov::element::string);
+    std::ifstream fin(file_name, std::ios::binary);
+    fin.seekg(offset);
+    auto bytes_to_read = tensor.get_byte_size();
+    fin.read(static_cast<char*>(tensor.data()), bytes_to_read);
+    OPENVINO_ASSERT(static_cast<size_t>(fin.gcount()) == bytes_to_read,
+                    "Cannot read ",
+                    bytes_to_read,
+                    "bytes from ",
+                    file_name);
+}
+
+namespace {
+size_t get_element_count(const element::Type& type, const size_t mem_size) {
+    if (ov::element::is_split_bit_type(type)) {
+        constexpr size_t storage_unit_size = 24;
+        size_t integer_number_bytes = mem_size * 8 / storage_unit_size;
+        return integer_number_bytes / type.bitwidth();
+    } else {
+        return mem_size * 8 / type.bitwidth();
+    }
+}
+ov::Shape calc_static_shape_for_file(const std::filesystem::path& file_name,
+                                     const element::Type& element_type,
+                                     const PartialShape& shape,
+                                     size_t offset) {
+    auto partial_shape = shape;
+    auto rank = partial_shape.rank();
+    OPENVINO_ASSERT(rank.is_static(), "Rank cannot be dynamic");
+    std::vector<size_t> dynamic_dimension_numbers;
+    size_t slice_size = 1;
+    for (size_t id = 0; id < partial_shape.size(); ++id) {
+        if (partial_shape[id].is_dynamic()) {
+            dynamic_dimension_numbers.push_back(id);
+        } else {
+            slice_size *= partial_shape[id].get_min_length();
+        }
+    }
+    OPENVINO_ASSERT(dynamic_dimension_numbers.size() == 1,
+                    "Only dynamic PartialShape with 1 dynamic dimension is supported");
+    auto& dynamic_dimension = partial_shape[dynamic_dimension_numbers[0]];
+
+    auto file_size = std::filesystem::file_size(file_name);
+    OPENVINO_ASSERT(file_size > offset, "Offset is bigger than size of file to read.");
+    auto elements_to_read = get_element_count(element_type, file_size - offset);
```

Comment:
This could be ov::Dimension and some properties can be re-used like div operator (if not possible to divide dimension will be invalid), use `compatible` dynamic_dimension.compatible(new_dimension_size)  etc.
It can be changed later. os not critical